Wikipedia/Code/WMFExploreSectionSchema.m

```diff
@@ -269,23 +270,43 @@ - (void)insertNearbySectionWithLocationIfNeeded:(CLLocation*)location {
     WMFExploreSection* oldNearby = [self existingNearbySection];
 
     // Check distance to old location
-    if (oldNearby.location && [location distanceFromLocation:oldNearby.location] < WMFMinimumDistanceBeforeUpdatingNearby) {
+    if (oldNearby.location && [location distanceFromLocation:oldNearby.location] < WMFMinimumDistanceBeforeUpdatingNearby && oldNearby.placemark != nil) {
         return;
     }
 
     // Check if already updated today
-    if (oldNearby.location && [oldNearby.dateCreated isToday]) {
+    if (oldNearby.location && [oldNearby.dateCreated isToday] && oldNearby.placemark != nil) {
         return;
     }
 
-    NSMutableArray<WMFExploreSection*>* sections = [self.sections mutableCopy];
-    [sections bk_performReject:^BOOL (WMFExploreSection* obj) {
-        return obj.type == WMFExploreSectionTypeNearby;
+    @weakify(self);
+    [self reverseGeocodeLocation:location completionHandler:^(CLPlacemark* _Nullable placemark) {
+        dispatchOnMainQueue(^{
+            @strongify(self);
+            NSMutableArray<WMFExploreSection*>* sections = [self.sections mutableCopy];
+            [sections bk_performReject:^BOOL (WMFExploreSection* obj) {
+                return obj.type == WMFExploreSectionTypeNearby;
+            }];
+
+            [sections wmf_safeAddObject:[self nearbySectionWithLocation:location placemark:placemark]];
+
+            [self updateSections:sections];
+        });
     }];
+}
 
-    [sections wmf_safeAddObject:[self nearbySectionWithLocation:location]];
+typedef void (^ WMFGeocodeCompletionHandler)(CLPlacemark* __nullable placemark);
 
-    [self updateSections:sections];
+- (void)reverseGeocodeLocation:(CLLocation*)location completionHandler:(nonnull WMFGeocodeCompletionHandler)completionHandler {
+    CLGeocoder* gc = [[CLGeocoder alloc] init];
+    [gc reverseGeocodeLocation:location completionHandler:^(NSArray < CLPlacemark* > * _Nullable placemarks, NSError* _Nullable error) {
```

Comment:
FYI, PromiseKit has a bunch of [CoreLocation categories](https://github.com/mxcl/PromiseKit/blob/master/Categories/CoreLocation/CLGeocoder%2BPromise.swift) that we can include via subspec if desired.